packages/studio-base/src/panels/ThreeDeeRender/Renderer.ts

```diff
@@ -1042,7 +1047,9 @@ export class Renderer extends EventEmitter<RendererEvents> implements IRenderer
     const t = translation;
     const q = rotation;
 
-    const transform = new Transform([t.x, t.y, t.z], [q.x, q.y, q.z, q.w]);
+    const transform = this.#transformPool.acquire();
+    transform.setPositionValues(t.x, t.y, t.z);
```

Comment:
Maybe drop the `Values` part of the name?

```suggestion
    transform.setPosition(t.x, t.y, t.z);
    transform.setRotation(q.x, q.y, q.z, q.w);
```